califa/calibration/R3BCalifaCrystalCal2Cluster.h

```diff
@@ -42,193 +44,101 @@ class R3BCalifaCrystalCal2Cluster : public FairTask
-    {
-        fClusterAlgorithmSelector = CONE;
-        fDeltaAngleClust = xDeltaAngleClust;
-    }
-
-    /** Public method SetPetalAlg
-     **
-     ** Select the petal clustering algorithm (2018 experiments)
-     **/
-    virtual void SetPetalAlg() { fClusterAlgorithmSelector = PETAL; }
+    void SetProtonClusterThreshold(Double_t clusterThresh){fProtonClusterThreshold = clusterThresh;}
 
-    /** Public method SetCrystalThreshold
-     **
-     ** Defines the minimum energy requested in a crystal to be considered in a calorimeter Hit
-     *@param thresholdEne  Double parameter used to set the threshold
-     **/
-    void SetCrystalThreshold(Double_t thresholdEne);
+    void SetSaturationEnergy(Double_t saturation) {fSaturation = saturation;}
 
-    /** Public method SetDRThreshold (for double reading)
-     **
-     ** Defines the minimum energy requested in a crystal to be considered in a calorimeter Hit
-     *@param thresholdEne  Double parameter used to set the threshold
-     **/
-    void SetDRThreshold(Double_t DRthresholdEne);
 
     /** Virtual method SetParContainers **/
     virtual void SetParContainers();
 
     /** Accessor to select online mode **/
     void SetOnline(Bool_t option) { fOnline = option; }
 
-    static int AngleToPetalId(TVector3& vec)
-    {
-        // internal [double] petal number, not corresponding to anything real
-        // only relevant for 2019 beam times
-        auto a = vec.Phi();
-        if (a < 0)
-        {
-            if (a < -2)
-                return 0;
-            else
-                return 1;
-        }
-        else
-        {
-            if (a < 2)
-            {
-                if (a < 0.2)
-                    return 2;
-                else
-                    return 3;
-            }
-            else
-                return 4;
-        }
-    }
 
     /** Virtual method Init **/
     virtual InitStatus Init();
 
     /** Virtual method ReInit **/
     virtual InitStatus ReInit();
 
-    void SetRandomization(Bool_t rand) { fRand = rand; }
+    void SetRandomization(Bool_t rand){fRand=rand;}
+
+    void SetRandomizationFile(TString file) {
 
-    void SetRandomizationFile(TString file)
-    {
+      fRandFile = file;
+      fHistoFile = new TFile(fRandFile);
 
-        fRandFile = file;
-        fHistoFile = new TFile(fRandFile);
     }
 
-  protected:
-    /** Method GetAnglesVector (calls R3BCalifaGeometry::GetAngles(id)) **/
-    TVector3 GetAnglesVector(int id);
-    TVector3 fTargetPos;
-    TVector3 fCalifaPos;
-    TVector3 fCalifatoTargetPos;
-
-    // Clustering algorithm selector
-    enum
-    {
-        INVALID = 0,
-        RECT = 1,
-        ROUND = 2,
-        ROUND_SCALED = 3,
-        CONE = 4,
-        ALL = 5,
-        NONE = 6,
-        PETAL = 7
-    } fClusterAlgorithmSelector; // Clustering algorithm selector
+
+    void IsSimulation(Bool_t simu) {fSimulation = simu;}
+
+    void SetTotalCrystals(Int_t total) {fTotalCrystals = total;}
```

Comment:
fTotalCrystals has a deeper meaning because it can be used to determine if a crystalId belongs to the low gain or high gain range. 
In theory, this should be strictly a function of R3BCalifaGeometry (or some mapping parameter interface), but in practice R3BCalifaGeometry is a semi-singleton with delayed initialization.

Reply:
Number of crystals is a variable used for selecting between two ranges in preamps with double reading. It's also useful to choose the best readout when you have two options.

Reply:
It looks this number should be a rather constant than a variable. Is this used for the compatibility for the very first campaigns for the demonstrator-era? If so, it can be kept like that I assume.